### DISCUSSION TITLE
How to cache fields that use @skip directives?

### DISCUSSION THREAD

**Discussion by gopeter** [urql-graphql/urql#1038] [Q&A] [answered]
I am building a large list of products where I can choose which columns I want to show:

```graphql
query GetArticleList(
  $skipSomeColumn: Boolean = false
) {
  articles {
    nodes {
      title
      description
      ... @skip(if: $skipSomeColumn) {
        someColumn
      }
    }
  }
}
```

The query works fine and returns only the data that the user wants to see. But I if navigate around the app and come back to the list (after the first successful fetch), all skippable fields are `undefined`.

Is there a way to configure `graphcache` to cache fields with these directives too?

**Comment by kitten** [comment, marked as answer]
Hm that's really odd. Graphcache is supposed to respect `@skip` just like GraphQL APIs do, so if the data isn't being skipped then it should be cached. We may need a small reproduction here to speed things along since we do have tests covering this case 🤔 

That being said, is this a type? I've never seen `@skip` being added to an inline fragment that isn't on a type, so I was wondering whether that's actually valid as-is in standard GraphQL

**Reply by kitten** [reply]
I just double checked the spec and our code; this is covered in the spec and is supported by Graphcache, so I believe I'll need some more information here 😅 

- The skip/include check: https://github.com/FormidableLabs/urql/blob/8e7fc5ce1451b24f569343011da717362c1ba3f0/exchanges/graphcache/src/operations/shared.ts#L116-L118
- The implementation of `shouldInclude`: https://github.com/FormidableLabs/urql/blob/8e7fc5ce1451b24f569343011da717362c1ba3f0/exchanges/graphcache/src/ast/traversal.ts#L44-L67

**Reply by kitten** [reply]
Ah ok, no, I believe it’s a bug on our end. I just realised that we don’t cover the case of inline fragments without a type name. I thought that was illegal syntax so I think I didn’t add a test case for this back when we wrote it.

So for the time being you can add the directives inline on fields or a separate fragment until we ship a patch soon.

Edit: Here we go, you can track the resolution of this on this issue: https://github.com/FormidableLabs/urql/issues/1039

**Comment by gopeter** [comment]
Wow, that was fast! 😄 I wanted to build a test case today to create a proper issue, but it seems that everything is clear now ☺️
